schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/application/ApplicationData.java

```diff
@@ -62,6 +63,16 @@ public Application getApplication() {
         return (T) application.getProperties().get(key);
     }
 
+    public <T> T getPropertyWithDefaultValue(String key, T defaultValue) {
+        T value = getProperty(key);
+        if (value == null || StringUtils.isBlank(value.toString())) {
+            value = defaultValue;
+            log.warn("No value defined (key: {}) defaulting to: {}", key, defaultValue);
+            setProperty(key, defaultValue);
+        }
+        return value;
+    }
```

Comment:
I really like the idea of the `getPropertyWithDefaultValue` method :+1: 
I actually thought: "Why didn't I do this before!?"
But checking if the value.toString is empty seems wrong to me as it makes it impossible to set an empty value for a property.